test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+#include "FakeTimeStampDataGenerator.hpp"
+
+#include "appfwk/app/Nljs.hpp"
+
+#include "logging/Logging.hpp"
+
+#include <chrono>
+#include <cstdlib>
+#include <thread>
+#include <string>
+#include <vector>
+
+using pd_clock = std::chrono::duration<double, std::ratio<1, 50000000>>;
+
+namespace dunedaq {
+namespace trigger {
+
+FakeTimeStampDataGenerator::FakeTimeStampDataGenerator(const std::string& name)
+  : dunedaq::appfwk::DAQModule(name)
+  , thread_(std::bind(&FakeTimeStampDataGenerator::do_work, this, std::placeholders::_1))
+  , outputQueue_()
+  , queueTimeout_(100)
+  , generator()
+{
+  register_command("conf",  &FakeTimeStampDataGenerator::do_configure);
+  register_command("start", &FakeTimeStampDataGenerator::do_start);
+  register_command("stop",  &FakeTimeStampDataGenerator::do_stop);
+  register_command("scrap", &FakeTimeStampDataGenerator::do_scrap);
+}
+
+void
+FakeTimeStampDataGenerator::init(const nlohmann::json& init_data)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+  auto ini = init_data.get<appfwk::app::ModInit>();
+  for (const auto& qi : ini.qinfos) {
+    if (qi.dir != "output") {
+      continue;                 // skip all but "output" direction
+    }
+    try
+    {
+      //outputQueue_.emplace_back(new sink_t(qi.inst));
+      outputQueue_.reset(new sink_t(qi.inst));
+    }
+    catch (const ers::Issue& excpt)
+    {
+      throw dunedaq::dunetrigger::InvalidQueueFatalError(ERS_HERE, get_name(), qi.name, excpt);
+    }
+  }
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting init() method";
+}
+
+void
+FakeTimeStampDataGenerator::get_info(opmonlib::InfoCollector& /*ci*/, int /*level*/)
+{
+}
+
+void
+FakeTimeStampDataGenerator::do_configure(const nlohmann::json& /*obj*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_configure() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_configure() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_start(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_start() method";
+  thread_.start_working_thread();
+  TLOG() << get_name() << " successfully started";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_start() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_stop(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_stop() method";
+  thread_.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_stop() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_scrap() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_scrap() method";
+}
+
+ std::vector<triggeralgs::TimeStampedData>
```

Comment:
Since we always return exactly one `TimeStampedData`, let's make the return type just `TimeStampedData`, not a vector. The function name `GetTimestamp` is a bit misleading, since it gets a TSD, not a timestamp. How about `get_time_stamped_data` (which also follows the `snake_case` convention from the style guide)